shared/api.ts

```diff
@@ -170,6 +184,21 @@ export const CACHED_API = {
       fetchCall: API.fetchIssuerWellknownConfig.bind(null, issuerId),
     }),
 
+  fetchIssuerAuthorizationServerMetadata: (
+    authorizationServerUrl: string,
+    isCachePreferred: boolean = false,
+  ) =>
+    generateCacheAPIFunction({
+      isCachePreferred,
+      cacheKey: API_CACHED_STORAGE_KEYS.fetchIssuerWellknownConfig(
```

Comment:
this cache key is same as of fetch issuer wellknown.

Reply:
Modified cache key to be specific to issuer authorization server metadata